test/sagemaker_tests/pytorch/__init__.py

```diff
@@ -0,0 +1,182 @@
+        source_image_uri.replace(image_region, target_region)
+        .replace(image_repo_name, target_image_repo_name)
+        .replace(image_account_id, target_account_id)
+    )
+
+    client = boto3.client('ecr', region_name = image_region)
+    username, password = get_ecr_login_boto3(client, image_account_id, image_region)
+    save_credentials_to_file(ECR_PASSWORD_FILE_PATH, password)
+
+    # using ctx.run throws error on codebuild "OSError: reading from stdin while output is captured".
+    # Also it throws more errors related to awscli if in_stream=False flag is added to ctx.run which needs more deep dive
+    subprocess.check_output(f"cat {ECR_PASSWORD_FILE_PATH} | docker login -u {username} --password-stdin https://{image_account_id}.dkr.ecr.{image_region}.amazonaws.com && docker pull {source_image_uri}", shell=True, executable="/bin/bash")
+    subprocess.check_output(f"docker tag {source_image_uri} {target_image_uri}", shell=True, executable="/bin/bash")
+    delete_file(ECR_PASSWORD_FILE_PATH)
+    username, password = get_ecr_login_boto3(target_ecr_client, target_account_id, target_region)
+    save_credentials_to_file(ECR_PASSWORD_FILE_PATH, password)
+    subprocess.check_output(f"cat {ECR_PASSWORD_FILE_PATH} | docker login -u {username} --password-stdin https://{target_account_id}.dkr.ecr.{target_region}.amazonaws.com && docker push {target_image_uri}", shell=True, executable="/bin/bash")
+    delete_file(ECR_PASSWORD_FILE_PATH)
+
+    return target_image_uri
+
+def get_ecr_image(ecr_image, region):
+    """
+    It uploads image to n_virginia region and return image uri
+    """
+    image_repo_uri, image_tag = ecr_image.split(":")
+    _, image_repo_name = image_repo_uri.split("/")
+    target_image_repo_name = f"{image_repo_name}"
+    regional_ecr_image = reupload_image_to_test_ecr(ecr_image, target_image_repo_name, region)
+    return regional_ecr_image
+
+
+def invoke_pytorch_helper_function(ecr_image, sagemaker_regions, helper_function, helper_function_args):
+    """
+    Used to invoke SM job defined in the helper functions in respective test file. The ECR image and the sagemaker session are passed explictly 
+    depending on the AWS region. 
+   This function will rerun for all SM regions after a defined wait time if capacity issues are seen.
+
+    :param ecr_image: ECR image in us-west-2 region
+    :param sagemaker_regions: List of SageMaker regions
+    :param helper_function: Function to invoke
+    :param helper_function_args: Helper function args 
+
+    :return: None
+    """
+
+    RETRY = 2
+    DELAY = 300
+    for _ in range(RETRY):
+        for region in sagemaker_regions:
+            sagemaker_session = get_sagemaker_session(region)
+            ecr_image = get_ecr_image(ecr_image, region) if region != "us-west-2" else ecr_image
+            try:
+                helper_function(ecr_image, sagemaker_session, **helper_function_args)
+                return
+            except Exception as e:
```

Comment:
why not:

```
except sagemaker.exceptions.UnexpectedStatusException as e:
    if "CapacityError" in str(e):
        continue
    else:
        raise
```

Reply:
this looks good as well, adding the change.